test/integration/test_config_defaults.py

```diff
@@ -161,7 +161,7 @@ def expected_default_config_dir(value):
     'conda_auto_init',  # broken: default overridden
     'config_dir',  # value overridden for testing
     'data_dir',  # value overridden for testing
-    'data_manager_config_file',  # broken: remove 'config/' prefix from schema
+    'data_manager_config_file',  # value overidden for testing
```

Comment:
Can this whole test go? Are there comparable unit tests at this point?

Having to explicitly place everything we override in here is sort of gross and unexpected - also it clashes with GALAXY_CONFIG_OVERRIDE_ and such for ad hoc testing - which I guess could break all the integration tests, but yeah...

Reply:
Yes, it could go. If you think it doesn't serve its purpose or causes more problems than it solves, I don't mind dropping it (I'd still test the schema defaults though). That said, a few arguments:
1. It's for config options defined in the schema *only*. It's not for everything we override (or create) in the config module.
2. We load these options automatically (which is tested), but we do not test overriding them. The logic for overriding options varies, and in some cases we were doing it wrong (`parse_config_file _options`, as per #9973). This test should prevent such bugs.
3. At least in theory, this unit test should not affect any integration or other unit tests.

That said, I do agree it's ugly. I tried generalizing it and setting new values automatically based on data type, but there are  too many special cases - I couldn't come up with an elegant solution. So it's not elegant. If the test also feels unexpected, that's worse (I can't judge that myself - I'm too "used to it").

To sum up, if you still feel it's better dropped, I'll drop it. In that case, I think I'll write a few tests targeting specific individual options (like the various config files).

Reply:
I'll remove the tests generated for overriding schema defaults before opening this for review. Keeping them for now while I ensure all other tests pass (they do pass locally).